File: scripts/lib/atproto/clients/blobs.py
from __future__ import annotations

from ..transport import TransportLayer


class BlobsClient:
    def __init__(self, transport: TransportLayer):
        self._t = transport

    def upload_blob(self, data: bytes, content_type: str, token: str) -> dict:
        return self._t.post_raw(
            "com.atproto.repo.uploadBlob", data, content_type, token=token
        )

File: scripts/lib/deno/clients/blobs.ts
import { TransportLayer } from "../transport.ts";

export class BlobsClient {
  constructor(private transport: TransportLayer) {}

  async uploadBlob(data: Uint8Array, contentType: string, token: string) {
    return await this.transport.postBinary("com.atproto.repo.uploadBlob", data, contentType, token);
  }
}
